zebra/zapi_msg.c

```diff
@@ -1869,8 +1871,29 @@ static int zapi_nhg_decode(struct stream *s, int cmd, struct zapi_nhg *api_nhg)
 				  "%s: Nexthop creation failed", __func__);
 			return -1;
 		}
+
+		if (max_weight < znh->weight) {
+			if (i != 0)
+				same_weight = false;
+
+			max_weight = znh->weight;
+		}
 	}
 
+	if (!same_weight) {
+		for (i = 0; i < api_nhg->nexthop_num; i++) {
+			uint64_t tmp;
+
+			znh = &((api_nhg->nexthops)[i]);
+
```

Comment:
it'd be clearer to have a comment in here, explaining the point of the arithmetic, shrinking the range of values in this way

Reply:
sure